src/Sentry/Infrastructure/DiagnosticLogger.cs

```diff
@@ -0,0 +1,48 @@
+using System;
+using System.Linq;
+using Sentry.Extensibility;
+
+namespace Sentry.Infrastructure
+{
+    /// <summary>
+    /// Base class for diagnostic loggers.
+    /// </summary>
+    public abstract class DiagnosticLogger : IDiagnosticLogger
```

Comment:
i think moving methods to a base class is a runtime breaking change.

Reply:
Not according to https://docs.microsoft.com/dotnet/core/compatibility/#members

> - ✔️ ALLOWED: Moving a member into a class higher in the hierarchy than the type from which it was removed

Unless I'm misinterpreting?

Reply:
consider me educated.